Docs: Grafana SCIM provisioning overview

vtorosyan opened the pull request:
**What is this feature?**

We are introducing SCIM provisioning in Grafana. This change is adding a baseline navigation tree as well as the landing page for SCIM provisioning.

You can review the changes visually here `<grafana_url>/docs/grafana/latest/setup-grafana/configure-security/configure-scim-provisioning/`

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/102170 https://github.com/grafana/grafana/issues/102175

**Special notes for your reviewer:**

**Please don't merge**. We are going to have other PRs and use this PR as an upstream to only merge all docs together when we are fully ready. That said, the changes here at the moment **do require a review**.

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

vtorosyan committed:
Docs: Create a landing page for SCIM provisioning

vtorosyan committed:
Fix docs warnings

irenerl24 commented:
@jdbaldry for you to review technically :)

jdbaldry commented on docs/sources/setup-grafana/configure-security/configure-scim-provisioning/_index.md:
```diff
@@ -0,0 +1,122 @@
+---
+aliases:
+  - ../../auth/scim/
+  - ../../enterprise/scim/
```
Aliases are for redirects which I don't believe we need since this is a new page

https://grafana.com/docs/writers-toolkit/write/front-matter/#aliases

```suggestion
```

jdbaldry commented on docs/sources/setup-grafana/configure-security/configure-scim-provisioning/_index.md:
```diff
@@ -0,0 +1,122 @@
+aliases:
+  - ../../auth/scim/
+  - ../../enterprise/scim/
+description: Learn how to use SCIM provisioning to synchronize users and groups from your identity provider to Grafana.
```
Consider adding a bit more description here. Generally aim for at least 150 characters (it's currently 106).

https://grafana.com/docs/writers-toolkit/write/front-matter/#aliases

jdbaldry commented on docs/sources/setup-grafana/configure-security/configure-scim-provisioning/_index.md:
```diff
@@ -0,0 +1,122 @@
+
+System for Cross-domain Identity Management (SCIM) is an open standard that allows automated user provisioning and management. With SCIM, you can automate the provisioning of users and groups from your identity provider to Grafana.
+
+> **Note:** Available in [Grafana Enterprise](../../../introduction/grafana-enterprise/) and [Grafana Cloud Advanced](/docs/grafana-cloud/).
```
```suggestion
{{< admonition type="note" >}}
Available in [Grafana Enterprise](../../../introduction/grafana-enterprise/) and [Grafana Cloud Advanced](/docs/grafana-cloud/).
{{< /admonition >}}
```

jdbaldry commented on docs/sources/setup-grafana/configure-security/configure-scim-provisioning/_index.md:
```diff
@@ -0,0 +1,122 @@
+
+> **Note:** Available in [Grafana Enterprise](../../../introduction/grafana-enterprise/) and [Grafana Cloud Advanced](/docs/grafana-cloud/).
+
+{{% admonition type="note" %}}
+This feature is behind the `enableSCIM` feature toggle.
+You can enable feature toggles through configuration file or environment variables. See configuration [docs](/docs/grafana/<GRAFANA_VERSION>/setup-grafana/configure-grafana/#feature_toggles) for details.
+{{% /admonition %}}
```
- [The percent syntax for admonitions is discouraged](https://grafana.com/docs/writers-toolkit/review/lint-prose/rules/#grafanashortcodes)
- [Prefer "refer to" instead of "see"](https://grafana.com/docs/writers-toolkit/review/lint-prose/rules/#grafanareferto)
- [Write useful link text](https://grafana.com/docs/writers-toolkit/write/links/useful-link-text/)

```suggestion
{{< admonition type="note" >}}
This feature is behind the `enableSCIM` feature toggle.
You can enable feature toggles through configuration file or environment variables.

For more information, refer to the [feature toggles documentation](/docs/grafana/<GRAFANA_VERSION>/setup-grafana/configure-grafana/#feature_toggles).
{{< /admonition >}}
```

jdbaldry commented on docs/sources/setup-grafana/configure-security/configure-scim-provisioning/_index.md:
```diff
@@ -0,0 +1,122 @@
+- **Reduced administrative overhead**: Eliminate manual user management tasks and reduce the risk of human error.
+- **Enhanced security**: Automatically disable access when users leave your organization, ensuring proper access control and compliance with security policies.
+
+{{% admonition type="note" %}}
+SCIM provisioning currently works **only with SAML authentication**. Other authentication methods are not supported.
+{{% /admonition %}}
```
- [Don't use bold for emphasis](https://grafana.com/docs/writers-toolkit/write/style-guide/style-conventions/#bold)
- Keep documentation timeless by avoiding words like "currently". Currently also implies that this is planned to change. Readers will want to know when this is planned to change.

```suggestion
{{< admonition type="note" >}}
SCIM provisioning only works SAML authentication.
Other authentication methods aren't supported.
{{< /admonition >}}
```

jdbaldry commented on docs/sources/setup-grafana/configure-security/configure-scim-provisioning/_index.md:
```diff
@@ -0,0 +1,122 @@
+
+## Supported identity providers
+
+The SCIM provisioning feature in Grafana currently supports:
```
```suggestion
The SCIM provisioning feature in Grafana supports:
```

vtorosyan commented:
Note: it seems we are not moving ahead with group attribute sync, so I might need to remove it from the comparison table.

jdbaldry commented on docs/sources/setup-grafana/configure-security/configure-scim-provisioning/_index.md:
```diff
@@ -0,0 +1,122 @@
+2. Your identity provider sends SCIM requests to the Grafana SCIM API endpoint.
+3. Grafana processes these requests to create, update, or deactivate users and teams, and synchronize team memberships.
+
+## Comparing SCIM with other user sync methods
```
A noun phrase makes it clear that this section is conceptual and not a task for how a user might compare SCIM with other user sync methods.

```suggestion
## Comparison of SCIM with other user sync methods
```

jdbaldry commented on docs/sources/setup-grafana/configure-security/configure-scim-provisioning/_index.md:
```diff
@@ -0,0 +1,122 @@
+
+## Comparing SCIM with other user sync methods
+
+Grafana offers several methods for synchronizing users, teams, and roles. The table below compares SCIM with other synchronization methods to help you understand its advantages:
```
https://developers.google.com/style/word-list#below

```suggestion
Grafana offers several methods for synchronizing users, teams, and roles.
The following table compares SCIM with other synchronization methods to help you understand its advantages:
```

jdbaldry commented on docs/sources/setup-grafana/configure-security/configure-scim-provisioning/_index.md:
```diff
@@ -0,0 +1,122 @@
+
+- **Complete automation**: SCIM is the only method that fully automates user and team provisioning.
+- **Dynamic team creation**: Teams are created automatically based on identity provider groups, eliminating manual setup.
+- **Near real-time synchronization**: Changes in your identity provider are reflected in Grafana based on your identity provider synchronization schedule (consult your identity provider documentation for specific details).
```
Avoids passive voice

```suggestion
- **Near real-time synchronization**: Grafana reflects changes in your identity provider using your identity provider synchronization schedule.
  Refer to your identity provider documentation for specific details.
```

jdbaldry commented on docs/sources/setup-grafana/configure-security/configure-scim-provisioning/_index.md:
```diff
@@ -0,0 +1,122 @@
+
+### Key advantages of SCIM over other methods
+
+- **Complete automation**: SCIM is the only method that fully automates user and team provisioning.
+- **Dynamic team creation**: Teams are created automatically based on identity provider groups, eliminating manual setup.
+- **Near real-time synchronization**: Changes in your identity provider are reflected in Grafana based on your identity provider synchronization schedule (consult your identity provider documentation for specific details).
+- **Enterprise-ready**: Designed for large organizations with complex user management needs.
```
Consider making this list more parallel to help with readability: https://developers.google.com/tech-writing/one/lists-and-tables#keep_list_items_parallel

jdbaldry commented on docs/sources/setup-grafana/configure-security/configure-scim-provisioning/_index.md:
```diff
@@ -0,0 +1,122 @@
+
+### SCIM and Team Sync
+
+{{% admonition type="warning" %}}
+Do not enable both SCIM and Team Sync simultaneously. These methods can conflict with each other, leading to unexpected behavior.
+{{% /admonition %}}
```
```suggestion
{{< admonition type="warning" >}}
Don't enable both SCIM and Team Sync simultaneously.
These methods can conflict with each other, leading to unexpected behavior.
{{< /admonition >}}
```

vtorosyan committed:
Update docs/sources/setup-grafana/configure-security/configure-scim-provisioning/_index.md

Co-authored-by: Jack Baldry <jack.baldry@grafana.com>

vtorosyan committed:
Update docs/sources/setup-grafana/configure-security/configure-scim-provisioning/_index.md

Co-authored-by: Jack Baldry <jack.baldry@grafana.com>

vtorosyan committed:
Apply suggestions from code review

Co-authored-by: Jack Baldry <jack.baldry@grafana.com>

vtorosyan committed:
Address some of the feedback and simplify few parts

vtorosyan committed:
Remove repetitive wording

vtorosyan committed:
Don't use possessive form, again

vtorosyan committed:
Merge branch 'main' into vt/scim_base_docs

vtorosyan commented:
@jdbaldry I really appreciate your review and thoughtful feedback. I _think_ I have addressed your feedback and included  suggestions. I have few additional "simplifications" as well.

I intend to use this as base PR as mentioned in the description, as there is a lot of new docs that will come with the introduction of SCIM. I don't want to waste anyone's time to review the changes with every single commit - so I will reach out for a final look to the Docs team when we have all pieces together.

sarahleejane commented on docs/sources/setup-grafana/configure-security/configure-scim-provisioning/_index.md:
```diff
@@ -0,0 +1,127 @@
+
+| Sync Method                                                                    | Users | Teams | Roles | Automation | Key Benefits                                                             | Limitations                                                  | On-Prem | Cloud |
+| ------------------------------------------------------------------------------ | ----- | ----- | ----- | ---------- | ------------------------------------------------------------------------ | ------------------------------------------------------------ | ------- | ----- |
+| SCIM                                                                           | ✅    | ✅    | ⚠️    | Partial    | Complete user and team lifecycle management with automatic team creation | Requires SAML authentication; uses Role Sync for basic roles | ✅      | ✅    |
```
It says automation partial - but in the summary below, it says full automation

vtorosyan replied:
Good catch, claiming that "SCIM is the only method that fully automates user and team provisioning" is still technically correct, as only role sync is not included, I will think how can we make it clear and less conflicting.

sarahleejane replied:
I think you should say Full here - since for the ticked columns (user and teams), the automation is full - right?

sarahleejane commented on docs/sources/setup-grafana/configure-security/configure-scim-provisioning/_index.md:
```diff
@@ -0,0 +1,127 @@
+Grafana offers several methods for synchronizing users, teams, and roles.
+The following table compares SCIM with other synchronization methods to help you understand its advantages:
+
+| Sync Method                                                                    | Users | Teams | Roles | Automation | Key Benefits                                                             | Limitations                                                  | On-Prem | Cloud |
```
Could you explain what these columns mean?  I'm not sure what you mean.

vtorosyan replied:
As Grafana offers several methods for synchronizing users, teams, and roles, the table compares SCIM with other synchronization methods to help to understand its advantages:

- **Users**: Whether the method can synchronize user accounts 
- **Teams**: Whether the method can synchronize teams and memberships
- **Roles**: Whether the method can synchronize user roles 
- **Automation**: Level of automation provided 
- **Key Benefits**: Main advantages of using this method
- **Limitations**: Important restrictions or requirements
- **On-Prem**: Available for on-premises Grafana installations 
- **Cloud**: Available for Grafana Cloud installations

sarahleejane replied:
If I understand right, team sync doesn't sync teams, but it does sync team membership.  You could almost pull them out as two separate columns to illustrate the difference between using team sync, or syncing teams with SCIM.

But maybe the table gets too big indeed.

vtorosyan replied:
Good point, let me think how can we make this table better and clear.

sarahleejane commented on docs/sources/setup-grafana/configure-security/configure-scim-provisioning/_index.md:
```diff
@@ -0,0 +1,127 @@
+Grafana offers several methods for synchronizing users, teams, and roles.
+The following table compares SCIM with other synchronization methods to help you understand its advantages:
+
+| Sync Method                                                                    | Users | Teams | Roles | Automation | Key Benefits                                                             | Limitations                                                  | On-Prem | Cloud |
```
We could add more columns to highlight your key advantages section below:

- Near real-time sync
- Dynamically created teams

vtorosyan replied:
I think it would make the table too big to be honest and provide repetitive information, as you mentioned those benefits are mentioned below (and at the top as well), so if you don't mind, I'd prefer to leave this as it is. 

The primary goal of the table is to provide a visual comparison to the other methods, so that from a quick glance they can see what features are available in which sync method.

sarahleejane commented on docs/sources/setup-grafana/configure-security/configure-scim-provisioning/_index.md:
```diff
@@ -0,0 +1,127 @@
+| Sync Method                                                                    | Users | Teams | Roles | Automation | Key Benefits                                                             | Limitations                                                  | On-Prem | Cloud |
+| ------------------------------------------------------------------------------ | ----- | ----- | ----- | ---------- | ------------------------------------------------------------------------ | ------------------------------------------------------------ | ------- | ----- |
+| SCIM                                                                           | ✅    | ✅    | ⚠️    | Partial    | Complete user and team lifecycle management with automatic team creation | Requires SAML authentication; uses Role Sync for basic roles | ✅      | ✅    |
+| [Team Sync](../configure-team-sync/)                                           | ❌    | ✅    | ❌    | Partial    | Maps identity provider groups to Grafana teams                           | Requires manual team creation                                | ✅      | ✅    |
```
Can any of these syncs be used in combination with each other?  I get confused to see them in a table this way because I would have thought you might use team sync + group attribute sync + role sync for example.

vtorosyan replied:
Team sync, group attribute sync and role sync are serving different purposes. 

As we agreed, group attribute sync needs to be removed from here (will push in my next commit), so let's focus on Team Sync and Role Sync.

- Team Sync: Used to synchronize team membership between IdP and Grafana (Note: it does not create, delete or update teams, just memberships). 
- Role Sync: Used to sync user roles upon login from the IdP.

So Role Sync and Team Sync can be combined together. As explained in the "Important considerations" section (which is actually changed in [this PR](https://github.com/grafana/grafana/pull/102478)), Team Sync and SCIM can't be combined, but Role Sync and SCIM can be combined. 

Since this is in the SCIM docs, the main goal is to describe how different are those from SCIM - we can consider adding somewhere else, more generic comparison of all these methods in our docs (outside of the scope of this PR)

sarahleejane reviewed (commented):
Looks great - left a few comments.

vtorosyan commented:
I've got another one for Managing users and teams https://github.com/grafana/grafana/pull/102478

vtorosyan committed:
Merge branch 'main' into vt/scim_base_docs

vtorosyan committed:
Address the feedback from PR

vtorosyan committed:
Merge branch 'main' into vt/scim_base_docs

vtorosyan committed:
Docs: Manage users and teams via SCIM (#102478)

* Docs: Manage users and teams via SCIM

* Docs: Manage users and teams via SCIM

* Refactor managing users parts

* Add team sync docs

* Update _index.md

* Apply suggestions from code review

Co-authored-by: linoman <2051016+linoman@users.noreply.github.com>

* Clarify how user identity linking is working

* Fix formatting

* Docs: Configure SCIM with Azure and Okta (#102582)

* Docs: Configure SCIM with Azure and Okta

* Apply suggestions from code review

Co-authored-by: Misi <mgyongyosi@users.noreply.github.com>

---------

Co-authored-by: Misi <mgyongyosi@users.noreply.github.com>

---------

Co-authored-by: linoman <2051016+linoman@users.noreply.github.com>
Co-authored-by: Misi <mgyongyosi@users.noreply.github.com>

vtorosyan committed:
Merge branch 'main' into vt/scim_base_docs

vtorosyan committed:
Add a warning about migration from team sync

sarahleejane commented on docs/sources/setup-grafana/configure-security/configure-scim-provisioning/_index.md:
```diff
@@ -0,0 +1,86 @@
+| SCIM                                                                           | ✅    | ✅    | ⚠️    | Partial    | Complete user and team lifecycle management with automatic team creation | Requires SAML authentication; uses Role Sync for basic roles | ✅      | ✅    |
+| [Team Sync](../configure-team-sync/)                                           | ❌    | ✅    | ❌    | Partial    | Maps identity provider groups to Grafana teams                           | Requires manual team creation                                | ✅      | ✅    |
+| [Active LDAP Sync](../configure-authentication/enhanced-ldap/)                 | ✅    | ❌    | ❌    | Full       | Background synchronization of LDAP users                                 | Limited to LDAP environments                                 | ✅      | ❌    |
+| [Role Sync](../configure-authentication/saml#configure-role-sync)              | ❌    | ❌    | ✅    | Partial    | Maps basic roles to users                                                | Limited to basic roles only                                  | ✅      | ✅    |
```
Why is role sync partial?  Does an IDP update not auto-propogate the basic role to Grafana?
Same questions for org mapping?

vtorosyan replied:
I take them as Partial because it is only limited to basic roles. But maybe we can say that automation is full, but add warning sign and mention that it is limited to the basic roles 💡 I think that reflects the reality better.

vtorosyan replied:
<img width="778" alt="Screenshot 2025-03-27 at 12 57 18" src="https://github.com/user-attachments/assets/fceef9dd-a31a-4b70-a1ef-56dbe78a2604" />

@sarahleejane I have updated the table taking your feedback in multiple places into consideration, can you please check again and see if this reflects the reality better? thanks for the review!

sarahleejane commented on docs/sources/setup-grafana/configure-security/configure-scim-provisioning/manage-users-teams/_index.md:
```diff
@@ -0,0 +1,160 @@
+
+- **Automate user lifecycle** from creation to deactivation
+- **Manage existing users** by linking them with identity provider identities
+- **Synchronize team memberships** based on identity provider group assignments
```
Automate team lifecycle from creation to deactivation - would be good to call out the auto-creation and deletion of team?

sarahleejane commented on docs/sources/setup-grafana/configure-security/configure-scim-provisioning/manage-users-teams/_index.md:
```diff
@@ -0,0 +1,160 @@
+### SCIM vs Team Sync
+
+{{< admonition type="warning" >}}
+Do not enable both SCIM and Team Sync simultaneously as these methods can conflict with each other.
```
Ah but I thought we would recommend that:
Coming soon - support for migrating Team Sync to SCIM Group Sync.
We don't recommend migrating from Team Sync until this support is released.
You may want to benefit from SCIM User Sync already, and this is compatible alongside Team Sync.

sarahleejane replied:
Perhaps what you mean here is: do not enable both team provisioning with SCIM Group Sync?

sarahleejane replied:
Because currently this reads like, you can't use SCIM at all if you are using Team Sync.

vtorosyan replied:
>Because currently this reads like, you can't use SCIM at all if you are using Team Sync.

I can see that it is confusing. If you read just this section without looking the entire page, it can create confusion. However, this warning is under the "SCIM vs Team Sync" section, so is only applicable to teams and not users. 

That said, I acknowledge that not everyone will read the doc entirely, so I will change this to make sure it's clear that user sync can still be used.

sarahleejane commented on docs/sources/setup-grafana/configure-security/configure-scim-provisioning/manage-users-teams/_index.md:
```diff
@@ -0,0 +1,160 @@
+- If you prefer Team Sync, do not enable SCIM provisioning
+
+{{< admonition type="warning" >}}
+**Migration Considerations:** Currently, there is no official migration path or tooling available for transitioning from Team Sync to SCIM. Attempting to migrate without proper tooling can lead to service disruption, loss of team permissions, and potential access issues. We recommend keeping your existing Team Sync setup until official migration tooling becomes available.
```
```suggestion
**Migration Considerations:** Currently, there is no official migration path or tooling available for transitioning from Team Sync to SCIM Group Sync. Attempting to migrate without proper tooling can lead to service disruption, loss of team permissions, and potential access issues. We recommend keeping your existing Team Sync setup until official migration tooling becomes available.
```

sarahleejane commented on docs/sources/setup-grafana/configure-security/configure-scim-provisioning/manage-users-teams/_index.md:
```diff
@@ -0,0 +1,160 @@
+### SCIM vs Team Sync
+
+{{< admonition type="warning" >}}
+Do not enable both SCIM and Team Sync simultaneously as these methods can conflict with each other.
```
```suggestion
Do not enable both SCIM Group Sync and Team Sync simultaneously as these methods can conflict with each other.
```

sarahleejane reviewed (approved):
This is super comprehensive, thank you!

vtorosyan committed:
Remove the SAML setup instructions from SCIM docs

vtorosyan committed:
Update _index.md

vtorosyan committed:
Update _index.md

jdbaldry commented on docs/sources/setup-grafana/configure-security/configure-scim-provisioning/configure-scim-with-azuread/_index.md:
```diff
@@ -0,0 +1,124 @@
+
+Configure the following required attributes:
+
+| Azure AD Attribute | Grafana Attribute            |
+| ------------------ | ---------------------------- |
+| userPrincipalName  | userName                     |
+| mail               | emails[type eq "work"].value |
+| displayName        | displayName                  |
```
```suggestion
| Azure AD Attribute.  | Grafana Attribute              |
| -------------------- | ------------------------------ |
| `userPrincipalName`  | `userName`                     |
| `mail`               | `emails[type eq "work"].value` |
| `displayName`        | `displayName`                  |
```

jdbaldry commented on docs/sources/setup-grafana/configure-security/configure-scim-provisioning/manage-users-teams/_index.md:
```diff
@@ -0,0 +1,195 @@
+- **Automate user lifecycle** from creation to deactivation
+- **Manage existing users** by linking them with identity provider identities
+- **Synchronize team memberships** based on identity provider group assignments
+- **Maintain security** through automated deprovisioning
```
Yeah I think we really are missing this word. @irenerl24 Do you think this should be in the dictionary?

irenerl24 replied:
Yes, I think it makes sense to add it, it's used in the IAM context

jdbaldry replied:
https://github.com/grafana/writers-toolkit/pull/1124

vtorosyan committed:
Apply suggestions from code review

Co-authored-by: Jack Baldry <jack.baldry@grafana.com>
Co-authored-by: Sarah Constant <sarahleejane@users.noreply.github.com>

vtorosyan committed:
Merge branch 'main' into vt/scim_base_docs

vtorosyan committed:
Address PR feedback

vtorosyan committed:
Stop using possessive form

vtorosyan committed:
Add a note mentioning that users/teams can not be manually deleted

vtorosyan committed:
Update _index.md

vtorosyan committed:
Docs: SCIM configuration options (#103085)

vtorosyan committed:
Merge branch 'main' into vt/scim_base_docs

vtorosyan committed:
Add a section to clarify how SCIM works with other IdPs

linoman reviewed (approved):
Looks great!

vtorosyan committed:
Update _index.md

vtorosyan committed:
Update _index.md

vtorosyan committed:
Merge branch 'main' into vt/scim_base_docs
